src/test/java/com/epam/esm/service/GiftCertificateServiceTest.java

```diff
@@ -0,0 +1,258 @@
+package com.epam.esm.service;
+
+import com.epam.esm.persistence.entity.GiftCertificate;
+import com.epam.esm.persistence.util.search.GiftCertificateSearchFilter;
+import com.epam.esm.service.dto.GiftCertificateDto;
+import com.epam.esm.service.util.mapper.AbstractEntityMapper;
+import com.epam.esm.web.util.exception.EntityBadInputException;
+import com.epam.esm.persistence.dao.GiftCertificateDao;
+import com.epam.esm.service.giftCertificate.GiftCertificateServiceImpl;
+import com.epam.esm.web.util.exception.EntityNotFoundException;
+import com.epam.esm.web.util.pagination.PaginationFilter;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class GiftCertificateServiceTest {
+
+    @Mock
+    private GiftCertificateDao giftCertificateDao;
+
+    @Mock
+    private AbstractEntityMapper<GiftCertificateDto, GiftCertificate> giftCertificateMapper;
+
+    @InjectMocks
+    private GiftCertificateServiceImpl giftCertificateService;
+
+    @Test
+    void findAllGiftsWithoutSearchFilterParams() {
+        GiftCertificateSearchFilter giftSearchFilter = new GiftCertificateSearchFilter();
+        GiftCertificate giftCertificate = GiftCertificate.builder()
+                .name("Test1")
+                .description("Test_test1")
+                .price(BigDecimal.valueOf(100L))
+                .duration(Duration.ofDays(10))
+                .build();
+        GiftCertificateDto giftCertificateDto = GiftCertificateDto.builder()
+                .name("Test1")
+                .description("Test_test1")
+                .price(BigDecimal.valueOf(100L))
+                .duration(Duration.ofDays(10))
+                .build();
+        int paginationItems = 1000;
+        PaginationFilter paginationFilter = PaginationFilter.builder()
+                .items(paginationItems)
+                .build();
+
+        when(giftCertificateMapper.toDto(any())).thenReturn(giftCertificateDto);
```

Comment:
when() calls are part of the arrange step, so they should be at placed to the first code block of the test method. Second block should consist of an actual method call, move it here from the third (assert) block.